xpano/log/logger.cc

```diff
@@ -64,7 +64,7 @@ void BufferSinkMt::flush_() {}
 Logger::Logger() : sink_(std::make_shared<BufferSinkMt>()) {}
 
 void Logger::RedirectSpdlogOutput(
-    std::optional<std::filesystem::path> app_data_path) {
+  std::optional<std::filesystem::path> app_data_path) {
   std::vector<spdlog::sink_ptr> sinks;
   sink_->set_pattern("[%l] %v");
   sinks.push_back(sink_);
```

Comment:
Please initialize the log_dir_ member here after line 73.

`log_dir_ = log_path.parent_path().string()`

https://en.cppreference.com/w/cpp/filesystem/path/parent_path

The member will be of type `std::optional<std::string>`, because it is possible that it will remain unset (if InitializePrefPath failed to return a value).

Then add an accessor function that you can call in the PanoGui constructor.